README.md

```diff
@@ -1,5 +1,162 @@
+
+        group.get_balances()  
+        -> { "Max": 0, "Ron": 0 }
+
+  - **ledger**  
+    'ledger' keeps the record of all transactions that have occurred in the group in the form of a specific namedtuple [Transaction](#transaction). The ledger can be retrieved as follows:
+
+        group.get_ledger()  
+        -> [Transaction(payer='Ernie', payees=['Allen'], amount=20, weights=None), Transaction(payer='Ellie', payees=['Joel', 'Allen'], amount=42, weights=[3, 4])]
+
+- Inheritence  
+  The group class inherits [collections.abc.Collection](https://docs.python.org/3/library/collections.abc.html#collections.abc.Collection). Consequently, the group objects will have `__len__`, `__contains__`, and `__iter__` attributes. The implementation of functions related to that are modified.  
+  The length of the object would return the number of members in the group. One can get the iterator for the keys of balances attribute of the group through `__iter__`. Also, one can skip using 'has_member' function, explained in the GroupFunctions section to check if a member exists in a group, or iterate through it as any iterable.
+
+      if 'Dom' in group:
+        # do something
+      for mem in group:
+        # member
+
+
+## Group Functions
+
+- **add_member ( string )**  
+  'add_member' adds a single individual to the group. No return value.
+
+      group.add_member('Sonya')  
+
+- **extend_members ( iterable of strings )**  
+  'extend_members' accepts an iterable of individuals to add to the group. No return value.
+
+      group.extend_members(['Jay', 'Matt'])
+
+- **getters**
+  - get_members: returns a list of all the members in the group  
+
+        group.get_members()
+  
+  - get_balances: returns a dictionary with members of the group mapping to their corresponding net balance in the group
+
+        group.get_balances()
+  
+  - get_balance(string): returns the net balance of a single member in the group. Raises an error if the member doesn't exist in the group
+
+        group.get_balance('Cathy')  
+        -> -5.6  
+  
+  - get_ledger: returns the ledger of the group, which is a list of [Transactions](#transaction)
+
+        group.get_ledger()  
+
+- **has_member ( string )**  
+  'has_member' returns true if the individual exists in the group, and false if not.
+
+        group.has_member('Pam')  
+        -> False
+
+- **clear**  
+  'clear' removes all the members from the group and all records from the ledger. It refreshes the group. No return value.
+
+      group.clear()
+
+- **remove_settled_member ( string )**
+  'remove_settled_member' does exactly as the name suggests. It accepts a member string and removes it from the group is the member is settled, which is if the corresponding net balance is zero. Removing someone yet to settle would make it impossible to settle the rest of the group. No return value.
```

Comment:
'remove_settled_member' does exactly as the name suggests. It accepts a member string and removes it from the group only if the member is settled, meaning the member's net balance is zero. Otherwise, the method raises a ValueError, since removing someone yet to settle would make it impossible to settle the rest of the group. No return value.